File: Data Structures & Algorithms/validate-parentheses/submission-2.py
from collections import deque
class Solution:
    def isValid(self, s: str) -> bool:
        stack = deque()

        for par in s:
            if par in "({[":
                stack.append(par)
            else:
                if not stack:
                    return False
                if par == ')' and stack[-1] != '(':
                    return False
                if par == '}' and stack[-1] != '{':
                    return False
                if par == ']' and stack[-1] != '[':
                    return False
                stack.pop()
            
        
        if not stack:
            return True
        else:
            return False


        

File: Data Structures & Algorithms/daily-temperatures/submission-2.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        
        std::stack<pair<int,int>> s;
        vector<int> res(temperatures.size(),0);

        for (int i = 0; i < temperatures.size();i++) {
            while (!s.empty() && temperatures[i] > s.top().first) {
                res[s.top().second] = i - s.top().second;
                s.pop();
            }
            s.push({temperatures[i],i});
        }

      /*
      [38,30,36]
      */  
      return res;
    }
};
